tests/freertos/semaphore_test.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019 Julian Holzwarth <julian.holzwarth@fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "freertos/semphr.h"
+#include "freertos/queue.h"
+#include "freertos/FreeRTOS.h"
+
+#include "semaphore_test.h"
+
+/*
+ * @brief amount of times the tested function will be called
+ */
+#define TEST_SEMAPHORE_FOR_COUNTER 20
+
+/**
+ * @brief   tests the take function for a freertos mutex semaphore
+ *
+ * @return pdPASS when the test is passed, pdFail otherwise
+ */
+int semaphore_test_mutex_take(void)
+{
+    SemaphoreHandle_t testing_semaphore = xSemaphoreCreateMutex();
```

Comment:
This is a memory leak since `xSemaphoreCreateMutex` mallocs a mutex object. If it is not freed before return, it keeps allocated.